### PR TITLE
Twilson63/chg convert map in memory 4

### PR DESCRIPTION
**Pull request by twilson63**
minisearch now uses sqlite as a data store to persist search documents...

### CODE REVIEW

**Review by TillaTheHun0 — COMMENTED** [review]
Implementation looks great. I think I found a typo and some superficial things, but all in all, looks really clean. Especially like the `dal` layer being injected.

File: dal.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// data access layer
+
+const store = `
+CREATE TABLE IF NOT EXISTS docs (
+  id TEXT,
+  type TEXT,
+  parent TEXT,
+  doc TEXT,
+  timestmp TEXT,
```
[the remark this answers is not in the document]
Comment (twilson63) [reply dal.js:9]: Old habit, time stamp used to be reserved word on some rdms

File: adapter.js [hunk 2]
```diff
@@ -43,48 +43,69 @@ const { allPass, keys, reduce } = R;
...
-    const store = new Map();
+    await db.post({
+      id: index,
+      type: "index",
+      parent: "root",
+      doc: {
+        id: index,
+        mappings,
+      },
+    });
```
[the remark this answers is not in the document]
Comment (twilson63) [reply adapter.js:92]: No reason started with promised and refactored to async just missed this one

**Comment by twilson63** [comment]
Thank you for the great review as I go through transitioning some of the other methods to Async, I had an idea, to wrap the miniSearch api with promise functions like I did with the dal and inject the minisearch abstraction into the adapter. This will allow for a cleaner flow with the logic I think, with both apis being promised based, converting them to Async and building pipelines will be much more clean and any errors thrown will be better controlled. It might take me another day to refactor, but I think it will be worth it.

File: sal.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  m.either(
+    (e) => Promise.reject({ ok: false, message: "SAL:" + e.message }),
+    (r) => Promise.resolve(r),
+  );
+
+const foldExists = (m) =>
+  m.either(
+    (e) => Promise.reject({ ok: false, message: "SAL:" + e.message }),
+    (r) => Promise.resolve(r ? true : false),
+  );
```
Comment (TillaTheHun0) [inline sal.js:22]: Could remove some repetition with something like:

```
const foldWith = (fn) => (m) =>
  m.either(
    (e) => Promise.reject({ ok: false, message: "SAL:" + e.message }),
    (_) => Promise.resolve(fn(_)),
  );

const fold = foldWith(constant({ ok: true }))

const foldIdentity = foldWith(identity)

const foldExists = foldWith(Boolean)
```

Up to you though.

**Review by TillaTheHun0 — DISMISSED** [review]
LGTM. Nice and clean. I just had one suggestion, but totally optional.

File: sal.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import { crocks, MiniSearch } from "./deps.js";
+
+const { tryCatch, compose, constant, identity  } = crocks;
+
+const foldWith = fn => m =>
+  m.either(
+    (e) => Promise.reject({ ok: false, message: "SAL:" + e.message }),
+    (_) => Promise.resolve(fn(_)),
+  )
+const fold = foldWith(constant({ok: true}))
```
Comment (TillaTheHun0) [inline sal.js:11]: don't think this is formatted with `deno lint`

Comment (twilson63) [reply sal.js:11]: Ugh

**Review by TillaTheHun0 — DISMISSED** [review]
Just notice some `deno fmt` issues with that last change. Once those are fixed, should be good to go.

**Review by TillaTheHun0 — APPROVED** [review]

**Pull request merged**